homeassistant/components/dsmr_reader/config_flow.py

```diff
@@ -0,0 +1,40 @@
+"""Config flow to configure DSMR Reader."""
+from __future__ import annotations
+
+from collections.abc import Awaitable
+import logging
+from typing import Any
+
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.config_entry_flow import DiscoveryFlowHandler
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def _async_has_devices(_: HomeAssistant) -> bool:
+    """MQTT is set as dependency, so that should be sufficient."""
```

Comment:
Can we not remove this then?

Reply:
No. If I remove it, the processing in the parent class fails as it evaluates to false by default, giving the error that no devices were found. I'm open for other solutions to that though, as I don't really like this either.

Reply:
Ok. See you return the parent class function where it's used.
I guess the better option is too keep it as you have it.